src/Compiler/TypedTree/TypedTreeOps.fs

```diff
@@ -9691,80 +9693,92 @@ let rec EvalAttribArgExpr (g: TcGlobals) x =
     | TypeOfExpr g _ -> x
     | TypeDefOfExpr g _ -> x
     | Expr.Op (TOp.Coerce, _, [arg], _) -> 
-        EvalAttribArgExpr g arg
+        EvalAttribArgExpr true g arg
     | EnumExpr g arg1 -> 
-        EvalAttribArgExpr g arg1
+        EvalAttribArgExpr true g arg1
     // Detect bitwise or of attribute flags
     | AttribBitwiseOrExpr g (arg1, arg2) -> 
-        let v1 = EvalAttribArgExpr g arg1
+        let v1 = EvalAttribArgExpr false g arg1
```

Comment:
Can you please replace the true/false argument with a more explicit option, to make it clear on the call-site?
Either a dedicated DU, or possibly a function passed in.